### PR TITLE
Auto-comment on submitted PRs

### PR DESCRIPTION
**Pull request by hsanjuan**
* Give a few tips on linking relevant discussions, self reviews, additional
  reviews and code contribution guidelines
* Be explicit about the next steps and what to expect.

I would like to trial this. A potential problem I see is that this will comment on every PR (including ours) and might be annoying for the people that sends most PRs. We can switch to commenting on only first-time PRs but then we miss the chance to remind people that only occasionally contribute (and ourselves about the process to follow).

### CODE REVIEW

**Comment by ribasushi** [comment]
@hsanjuan is it possible to give this some sort of predefined `Subject` or somesuch, so that one can auto-filter these out? Our mailboxes are overloaded as it is...

**Comment by hsanjuan** [comment]
> @hsanjuan is it possible to give this some sort of predefined `Subject` or somesuch, so that one can auto-filter these out? Our mailboxes are overloaded as it is...

You would be able to filter out the botname (even block it on github so you don't even see these).

File: .github/auto-comment.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+pullRequestOpened: >
+
+  Thank you for submitting this PR!
+
+  A maintainer will be here shortly to review it.
+
+  We are super grateful to get code contributions and aim to make the best of
+  the contribution process. However, we are also overloaded with work, we'd be
+  happy if you can make sure that:
```
Comment (Stebalien) [inline .github/auto-comment.yml:13]: The wording in the last sentence is a bit awkward.

File: .github/auto-comment.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+    * The context for this PR is cleared, with relevant discussion, decisions
```
Comment (Stebalien) [inline .github/auto-comment.yml:15]:
```suggestion
    * The context for this PR is clear, with relevant discussion, decisions
```

File: .github/auto-comment.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+      and stakeholders linked/mentioned.
+
+    * You have self-reviewed yourself. Please make a self-review and leave
```
Comment (Stebalien) [inline .github/auto-comment.yml:18]:
```suggestion
    * You have self-reviewed it yourself. Please make a self-review and leave
```

Comment (Stebalien) [reply .github/auto-comment.yml:18]: I'd also make it clear that _most_ motivation should go in code comments. Self-reviews are just for additional motivation related to the _change_ that wouldn't make much sense as a code comment.

**Comment by hsanjuan** [comment]
> This is a _lot_ of text for an automatic comment.

I know. I've done another pass trying to be more concise. Can you check?

**Pull request merged**